.github/workflows/ci.yaml

```diff
@@ -7,296 +7,110 @@ on:
     env:
-      DEVELOPER: 1
-      VALGRIND: 0
-      EXPERIMENTAL_FEATURES: 0
+      RUST: 1
       COMPAT: 1
     strategy:
       fail-fast: true
-      matrix:
-        include:
-          - CFG: "make and unit test w/ VALGRIND"
-            TEST_CMD: "make default check-source"
-            VALGRIND: 1
-          - CFG: "make-O3-check"
-            TEST_CMD: "make check-source check-units installcheck check-gen-updated"
-            COPTFLAGS: "-O3"
-          - CFG: "make-32-bit-nodev-check"
-            ARCH: 32
-            TEST_CMD: "make check-source check-units installcheck"
-            DEVELOPER: 0
-          - CFG: "make-EXPERIMENTAL-check"
-            TEST_CMD: "make check-source check-units installcheck check-gen-updated"
-            EXPERIMENTAL_FEATURES: 1
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v2.0.0
 
       - name: Set up Python 3.7
-        uses: actions/setup-python@v4
+        uses: actions/setup-python@v2
         with:
           python-version: 3.7
 
       - name: Install dependencies
         run: |
           bash -x .github/scripts/setup.sh
-
-      - name: Build
-        env:
-          VALGRIND: ${{ matrix.VALGRIND }}
-          DEVELOPER: ${{ matrix.DEVELOPER }}
-          EXPERIMENTAL_FEATURES: ${{ matrix.EXPERIMENTAL_FEATURES }}
-          COMPILER: ${{ matrix.COMPILER }}
-          ARCH: ${{ matrix.ARCH }}
-          COMPAT: ${{ matrix.COMPAT }}
-          PYTEST_PAR: ${{ matrix.PYTEST_PAR }}
-          PYTEST_OPTS: ${{ matrix.PYTEST_OPTS }}
-          COPTFLAGS: ${{ matrix.COPTFLAGS }}
-          NETWORK: ${{ matrix.NETWORK }}
-          TEST_CMD: ${{ matrix.TEST_CMD }}
-          TEST_GROUP_COUNT: ${{ matrix.TEST_GROUP_COUNT }}
-          TEST_GROUP: ${{ matrix.TEST_GROUP }}
+          pip install -U pip wheel poetry
+          # Export and then use pip to install into the current env
+          poetry export -o /tmp/requirements.txt --without-hashes --with dev
+          pip install -r /tmp/requirements.txt
+      - name: Run checks
         run: |
-          echo $PROTOC
-          which protoc
-          bash -x .github/scripts/build.sh
-
-  check-dock:
-    name: Check core-lightning doc
-    runs-on: ubuntu-20.04
-    env:
-      DEVELOPER: 1
-      VALGRIND: 0
-      EXPERIMENTAL_FEATURES: 0
-      COMPAT: 1
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-
-      - name: Set up Python 3.7
-        uses: actions/setup-python@v4
-        with:
-          python-version: 3.7
-
-      - name: Install dependencies
-        run: bash -x .github/scripts/setup.sh
-
-      - name: Check Doc
-        run: |
-          pip install mako
           ./configure
-          make check-doc
-
-  normal-test:
-    name: Normal Test Config ${{ matrix.cfg }}
+          make -j 4 check-source
+          make -j 4 check-units
+          make -j 4 check-gen-updated
+          make -j 4 check-doc
+          make -j 4 installcheck
```

Comment:
We could, but we may not be happy with the 2 we get here (4 is just a random value I found worked ok btw), since most tasks are not CPU bound but rather IO-bound it. We can likely tweak these values to get the shortest possible runtime.

I'm also not quite happy that some of these static checks actually end up building the binaries (`installcheck` obviously does). Ideally we'd have very lightweight checks that gate the heavier executions (and I'm counting building the binaries as heavy) and defer any heavy or time-intensive task to later stages.